backend/main/serializers/answer_serializer.py

```diff
@@ -1,14 +1,25 @@
 from rest_framework import serializers
 from ..models.answer import Answer
+from ..models.choice import Choice
+from ..models.question import Question
 
 class AnswerSerializer(serializers.ModelSerializer):
-    user_id = serializers.PrimaryKeyRelatedField(many=False, read_only=True)
-    question_id = serializers.PrimaryKeyRelatedField(many=False, read_only=True)
-    lesson_id = serializers.PrimaryKeyRelatedField(many=False, read_only=True)
-    choice_id = serializers.PrimaryKeyRelatedField(many=False, read_only=True)
-
     class Meta:
         model = Answer
         fields = '__all__'
 
+    def create(self, validated_data):
+        choice = Choice.objects.get(pk=self.data['choice'])
+        question = Question.objects.get(pk=self.data['question'])
+        is_correct = True if question.correct_answer == choice.value else False
 
+        answer = Answer(
+            user=validated_data['user'],
+            question=validated_data['question'],
+            lesson=validated_data['lesson'],
+            choice=validated_data['choice'],
+            is_correct=is_correct,
+        )
```

Comment:
Try removing `answer.save()` in line 24 then try this code below and see if it still works 
```suggestion
        answer = Answer.objects.create(
            user=validated_data['user'],
            question=validated_data['question'],
            lesson=validated_data['lesson'],
            choice=validated_data['choice'],
            is_correct=is_correct,
        )
```